### PR TITLE
Code generated using native OpenAPI specification

### PR DESCRIPTION
**Pull request by saimedhi**
### Description
Code generated using native OpenAPI specification

### Issues Resolved
Related to https://github.com/opensearch-project/opensearch-py/issues/636
Related to https://github.com/opensearch-project/opensearch-py/issues/474

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by saimedhi** [comment]
This is generated code. 

@dblock, please review. Kindly approve if everything looks correct, but please do not merge. I'll give it one last check to make sure we haven't missed any minor details. Thank you!

**Review by dblock — COMMENTED** [review]
This looks good, and it adds the notifications plugin support. Should be in CHANGELOG at least.

**Review by dblock — DISMISSED** [review]

**Comment by saimedhi** [comment]
> This looks good, and it adds the notifications plugin support. Should be in CHANGELOG at least.

In the follow-up PR, I'll update the generator to include any newly added namespaces in the changelog.

File: utils/templates/base [hunk 1]
```diff
@@ -22,7 +22,7 @@
         {% for p, info in api.params %}
         {% if info.description %}
         {% filter wordwrap(72, wrapstring="\n            ") %}
-        :arg {{ p }}{% if info.deprecated and info.deprecation_message is defined %} (Deprecated: {{ info['deprecation_message'][:-1] }}.){% endif %}: {{ info.description }} {% if info.options and "Valid values: " not in info.description %}Valid choices are {{ info.options|join(", ") }}.{% endif %}
+        :arg {{ p }}{% if info.deprecated and info.deprecation_message is defined %} (Deprecated: {{ info['deprecation_message'][:-1] }}.){% endif %}: {{ info.description }} {% if info.options and "Valid values" not in info.description %}Valid choices are {{ info.options|join(", ") }}.{% endif %}
         {% if info.default is defined %}{% if info.default is not none %}{% if info.default is sameas(false) %}Default is false.{% else %}Default is {{ info.default }}.{% endif %}{% endif %}{% endif %}
```
Comment (saimedhi) [inline utils/templates/base:26]: Made small modification to generator

**Review by dblock — DISMISSED** [review]

**Comment by saimedhi** [comment]
@dblock, apologies for the inconvenience. I need your approval one last time before merging. Thank you.

**Review by dblock — APPROVED** [review]

**Pull request merged**